packages/@aws-cdk/aws-codepipeline/lib/github-source-action.ts

```diff
@@ -44,9 +45,13 @@ export interface GitHubSourceActionProps extends actions.CommonActionProps,
   oauthToken: cdk.Secret;
 
   /**
-   * Whether or not AWS CodePipeline should poll for source changes
+   * Whether AWS CodePipeline should poll for source changes.
+   * If this is `false`, the Pipeline will use a webhook to detect source changes instead.
+   * Note that CodePipeline must have been given permissions to the GitHub account owning this repository
+   * in order for the webhook-based change detection to work.
+   * You can grant those permissions using the CodePipeline UI in the AWS Console.
    *
-   * @default true
+   * @default false
```

Comment:
That is an excellent point. Let me discuss this with the CodePipeline team. I'll circle back to this question.

Reply:
Any update on this?  Hooks are significantly faster to trigger the pipeline vs polling, but that's likely non-obvious to first-time users

Reply:
Also, doesn't the token step have to be done regardless to be able to poll and retrieve the GH repo?

Reply:
> Any update on this? Hooks are significantly faster to trigger the pipeline vs polling, but that's likely non-obvious to first-time users

Sorry, didn't have time to consult with the CodePipeline team last week. I have a meeting with them tomorrow, I'll report back what I find out.

> Also, doesn't the token step have to be done regardless to be able to poll and retrieve the GH repo?

You need a token, true, but for webhooks, there's an extra step - you need to give the CodePipeline OAuth app in region permissions to your account, which you don't need when using polling.

Reply:
Sounds like there should be some documentation around this at least on the `README.md` file here, too. Give users a clear walk-trough on how to get this set-up correctly.

Reply:
CodePipeline won't use a personal access token to create the webhook even if the token has admin:repo_hook permission?

Reply:
Yep, Clare is right. I've discussed this with the CodePipeline team, and tested myself. You don't need to give the CodePipeline OAuth application permissions if you're using a personal token - which you need anyway to use the GitHub Action, even if it uses polling instead of a webhook. So changing the default should be safe here.